com.vrcfury.vrcfury/Editor/VF/Feature/CombineMeshesBuilder.cs

```diff
@@ -0,0 +1,543 @@
+
+        public override string GetEditorTitle()
+        {
+            return "Combine Meshes";
+        }
+
+        public override VisualElement CreateEditor(SerializedProperty prop)
+        {
+            var content = new VisualElement();
+            content.Add(VRCFuryEditorUtils.Info(
+                "This feature will combine all similar meshes that are not animated to be toggled into one, " +
+                " saving extra draw calls!"
+            ));
+            content.Add(new VisualElement { style = { paddingTop = 10 } });
+            content.Add(VRCFuryEditorUtils.Prop(prop.FindPropertyRelative("keepFaceMeshSeparate"), "Keep Face Mesh separate"));
+            content.Add(new VisualElement { style = { paddingTop = 10 } });
+            return content;
+        }
+
+        public override bool AvailableOnProps()
+        {
+            return false;
+        }
+
+        [FeatureBuilderAction(FeatureOrder.BlendshapeOptimizer)]
+        public void Apply()
+        {
+            var changes = CalculateMeshChanges();
+            foreach (var group in changes.Groups)
+            {
+                var skins = group.Value;
+                var basis = skins[0];
+                VFGameObject basisObject = basis.gameObject;
+                var meshesToAdd = skins.Skip(1).ToList();
+                basis.sharedMesh = mutableManager.MakeMutable(basis.sharedMesh, basis.owner()); ;
+                VRCFuryEditorUtils.MarkDirty(basis);
+
+                var combinable = new CombinableMesh(basis);
+                foreach (var mesh in meshesToAdd)
+                {
+                    combinable.AddMesh(mesh);
+                    VFGameObject obj = mesh.gameObject;
+                    mover.DirectRewrite(obj, basisObject);
+                    obj.Destroy();
+                }
+                combinable.Combine();
+            }
+            foreach (var skin in changes.Removed)
+            {
+                VFGameObject obj = skin.gameObject;
+                obj.Destroy();
+            }
+        }
+
+        private struct MeshChanges
+        {
+            public Dictionary<SkinnedMeshRendererGroup, List<SkinnedMeshRenderer>> Groups { get; set; }
+            public List<SkinnedMeshRenderer> Toggled { get; set; }
+            public List<SkinnedMeshRenderer> Removed { get; set; }
+            public List<SkinnedMeshRenderer> NotCombinable { get; set; }
+        }
+
+        private MeshChanges CalculateMeshChanges()
+        {
+            var task = new MeshChanges();
+            task.Groups = new Dictionary<SkinnedMeshRendererGroup, List<SkinnedMeshRenderer>>();
+            task.Removed = new List<SkinnedMeshRenderer>();
+            task.Toggled = new List<SkinnedMeshRenderer>();
+            task.NotCombinable = new List<SkinnedMeshRenderer>();
+
+            var descriptor = avatarObject.GetComponent<VRC_AvatarDescriptor>();
```

Comment:
Use VRCAvatarDescriptor rather than VRC_AvatarDescriptor since it's 'technically' part of their public API (even though it's all sort of a mess)